File: backend/src/modules/plano/entity/Plano.js
class Plano {
  constructor({ id_plano, nome, valor, duracao_meses, descricao }) {
    this.id_plano = id_plano;
    this.nome = nome;
    this.valor = valor;
    this.duracao_meses = duracao_meses;
    this.descricao = descricao;
  }

  validate() {
    const errors = [];
    if (!this.nome) errors.push('Nome é obrigatório');
    if (!this.valor) errors.push('Valor é obrigatório');
    if (!this.duracao_meses) errors.push('Duração em meses é obrigatória');
    return errors;
  }
}

module.exports = Plano;

File: backend-spring/src/main/java/com/vitalativa/entity/Plano.java
package com.vitalativa.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Positive;

@Entity
@Table(name = "plano")
public class Plano {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_plano")
    private Integer idPlano;
    
    @NotBlank(message = "Nome é obrigatório")
    private String nome;
    
    @Positive(message = "Valor deve ser positivo")
    private Double valor;
    
    @Column(name = "duracao_meses")
    private Integer duracaoMeses;
    
    private String descricao;
    
    public Plano() {}
    
    public Integer getIdPlano() { return idPlano; }
    public void setIdPlano(Integer idPlano) { this.idPlano = idPlano; }
    public String getNome() { return nome; }
    public void setNome(String nome) { this.nome = nome; }
    public Double getValor() { return valor; }
    public void setValor(Double valor) { this.valor = valor; }
    public Integer getDuracaoMeses() { return duracaoMeses; }
    public void setDuracaoMeses(Integer duracaoMeses) { this.duracaoMeses = duracaoMeses; }
    public String getDescricao() { return descricao; }
    public void setDescricao(String descricao) { this.descricao = descricao; }
}